### DISCUSSION TITLE
Bing photo of the day

### DISCUSSION THREAD

**Discussion by jacobgad** [linuxserver/Heimdall#690] [Ideas]
set the background image to change everyday acording to bing picture of the day 

here is a link on how to do it:
https://stackoverflow.com/questions/10639914/is-there-a-way-to-get-bings-photo-of-the-day

**Comment by nomandera** [comment]
Similar to https://github.com/linuxserver/Heimdall/issues/69

Do we need multiple sources or is this just an viable alternative?

**Comment by jacobgad** [comment]
Here is a github page with all the code you will need to get it sorted.
Please let me know how you go

https://github.com/grubersjoe/bing-daily-photo/blob/master/README.md

**Comment by mhbates** [comment]
Has anyone figured out another way of doing this? Is it possible to make a script to pull the image of the day and give to Heimdall somehow?

**Comment by insanepoet** [comment]
I use a different provider but i use a bash script on cron to achieve nearly the same effect on a different project. Bash script grabs the photo, removes the old BGPhoto.jpg and mv the new photo to BGPhoto.jpg. Cron calls the script every morning at 0209. 

As long as you have a volume accessible to the Heimdall container you can run it locally and the container will access it from the root or you could setup cron in the container to run the script there. Bash script only requires curl (i mean do you not already have curl installed?)

Bash Script - https://gist.github.com/insanepoet/cf4074d7aa1a455ff5b0437497f8dd14

If you want to run it in the container you can read up on how here
Cron in container info - https://github.com/Ekito/docker-cron

For a more advanced script (not yet modified the way I have above) that puts the EXIF data in the lower right of the wallpaper, you could look at https://pastebin.com/iSDF10DK but it requires "exiftool" and "imagemagick"

This can certainly be achieved in php but it all depends on if this is an edge use case or an actual feature that should be implemented. (make sure you make the script executable)
